misk-redis/src/testFixtures/kotlin/misk/redis/testing/FakeRedis.kt

```diff
@@ -411,4 +422,96 @@ class FakeRedis @Inject constructor(
   override fun publish(channel: String, message: String) {
     throw NotImplementedError("Fake client not implemented for this operation")
   }
+
+  private fun zadd(
+    key: String,
+    score: Double,
+    member: ByteString,
+    options: Set<Redis.ZAddOptions>,
+  ): Long {
+    verifyZAddOptions(options)
+    var newFieldCount = 0L
+    var elementsChanged = 0L
+
+    if (!sortedSetKeyValueStore.containsKey(key)) {
+      sortedSetKeyValueStore[key] = Value(data = sortedMapOf(), expiryInstant = Instant.MAX)
+    }
+    val exists = sortedSetKeyValueStore[key]!!.data[member] != null
+
+    // if no options, overwrite or add
+    if (options.isEmpty() || (options.size == 1 && options.contains(CH))) {
+      sortedSetKeyValueStore[key]!!.data[member] = score
+      if (!exists) newFieldCount++
+      elementsChanged++
+    }
+
+    if (options.contains(NX) && !exists) { // only add new ones
+      sortedSetKeyValueStore[key]!!.data[member] = score
+      newFieldCount++
+      elementsChanged++
+    } else if ( // overwrite existing if score less than and add new ones.
+        (options.contains(LT) && exists && score < sortedSetKeyValueStore[key]!!.data[member]!!)
+        || (options.contains(LT) && !exists)
+      ) {
+      sortedSetKeyValueStore[key]!!.data[member] = score
+      if (!exists) newFieldCount++
+      elementsChanged++
+    } else if ( // overwrite existing if score greater than and add new ones.
+        (options.contains(GT) && exists && score > sortedSetKeyValueStore[key]!!.data[member]!!)
+        || (options.contains(GT) && !exists)) {
+      sortedSetKeyValueStore[key]!!.data[member] = score
+      if (!exists) newFieldCount++
+      elementsChanged++
```

Comment:
Is the behaviour corrrect if you do `zadd LT XX` or `zadd GT XX`? It looks like when `LT` or `GT` is set, `XX` gets ignored

Reply:
in `zadd LT XX` or `zadd GT XX` the contract is this.. 
```
 for existing ones, if the score is less/more than the existing scores - update
 XX will prevent adding new ones.
```
I have updated the code now. So this should be fixed.